sdk/nodejs/x/automation/localWorkspace.ts

```diff
@@ -0,0 +1,592 @@
+     *
+     * @param stackName The stack to refresh
+     */
+    async refreshConfig(stackName: string): Promise<ConfigMap> {
+        await this.selectStack(stackName);
+        await this.runPulumiCmd(["config", "refresh", "--force"]);
+        return this.getAllConfig(stackName);
+    }
+    /**
+     * Returns the currently authenticated user.
+     */
+    async whoAmI(): Promise<WhoAmIResult> {
+        const result = await this.runPulumiCmd(["whoami"]);
+        return { user: result.stdout.trim() };
+    }
+    /**
+     * Returns a summary of the currently selected stack, if any.
+     */
+    async stack(): Promise<StackSummary | undefined> {
+        const stacks = await this.listStacks();
+        for (const stack of stacks) {
+            if (stack.current) {
+                return stack;
+            }
+        }
+        return undefined;
+    }
+    /**
+     * Returns all Stacks created under the current Project.
+     * This queries underlying backend and may return stacks not present in the Workspace (as Pulumi.<stack>.yaml files).
+     */
+    async listStacks(): Promise<StackSummary[]> {
+        const result = await this.runPulumiCmd(["stack", "ls", "--json"]);
+        const stacks: StackSummary[] = JSON.parse(result.stdout);
+        return stacks;
+    }
+    /**
+     * serializeArgsForOp is hook to provide additional args to every CLI commands before they are executed.
+     * Provided with stack name,
+     * returns a list of args to append to an invoked command ["--config=...", ]
+     * LocalWorkspace does not utilize this extensibility point.
+     */
+    async serializeArgsForOp(_: string): Promise<string[]> {
+        // LocalWorkspace does not utilize this extensibility point.
+            return [];
+    }
+    /**
+     * postCommandCallback is a hook executed after every command. Called with the stack name.
+     * An extensibility point to perform workspace cleanup (CLI operations may create/modify a Pulumi.stack.yaml)
+     * LocalWorkspace does not utilize this extensibility point.
+     */
+    async postCommandCallback(_: string): Promise<void> {
+        // LocalWorkspace does not utilize this extensibility point.
+        return;
+    }
+    /** @internal */
+    private async runPulumiCmd(
+        args: string[],
+    ): Promise<CommandResult> {
+        let envs: { [key: string]: string } = {};
+        if (this.pulumiHome) {
+            envs["PULUMI_HOME"] = this.pulumiHome;
+        }
+        envs = { ...envs, ...this.envVars };
+        return runPulumiCmd(args, this.workDir, envs);
+    }
+}
+
+/**
+ * Description of a stack backed by an inline (in process) Pulumi program.
+ */
+export interface InlineProgramArgs {
+    /**
+     * The name of the associated Stack
+     */
+    stackName: string;
+    /**
+     * The name of the associated project
+     */
+    projectName: string;
+    /**
+     * The inline (in process) Pulumi program to use with Update and Preivew operations.
```

Comment:
Lil' nit I just noticed while testing.

```suggestion
     * The inline (in process) Pulumi program to use with Update and Preview operations.
```